submission/persistence-mongo/src/test/kotlin/ac/uk/ebi/biostd/persistence/doc/service/SubmissionDraftMongoServiceTest.kt

```diff
@@ -0,0 +1,112 @@
+package ac.uk.ebi.biostd.persistence.doc.service
+
+import ac.uk.ebi.biostd.persistence.common.request.PaginationFilter
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionQueryService
+import ac.uk.ebi.biostd.persistence.doc.db.data.SubmissionDraftDocDataRepository
+import ac.uk.ebi.biostd.persistence.doc.test.doc.DRAFT_CONTENT
+import ac.uk.ebi.biostd.persistence.doc.test.doc.DRAFT_KEY
+import ac.uk.ebi.biostd.persistence.doc.test.doc.USER_ID
+import ac.uk.ebi.biostd.persistence.doc.test.doc.testDocDraft
+import ebi.ac.uk.extended.model.ExtSubmission
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.junit5.MockKExtension
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.unmockkStatic
+import io.mockk.verify
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.extension.ExtendWith
+import uk.ac.ebi.extended.serialization.service.ExtSerializationService
+import java.time.Instant
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@ExtendWith(MockKExtension::class)
+internal class SubmissionDraftMongoServiceTest(
+    @MockK private val draftDocDataRepository: SubmissionDraftDocDataRepository,
+    @MockK private val submissionQueryService: SubmissionQueryService,
+    @MockK private val extSerializationService: ExtSerializationService
+) {
+    private val testInstance = SubmissionDraftMongoService(
+        draftDocDataRepository,
+        submissionQueryService,
+        extSerializationService
+    )
+
+    @Test
+    fun `get submission when exists`() {
```

Comment:
I think I had made a similar comment in a separate PR, not sure if this is just moving the same file to a different package or a very similar test but, in any case, I think the names should be a bit more descriptive. As far as I recall there are 4 scenarios:

1. The submission exists and there is no draft
2. The submission exists and there is a draft
3. There is a draft and the submission doesn't exist
4. Neither the submission nor the draft exists

I think the tests are there but the names are not very clear.